src/Init/Data/Array/Basic.lean

```diff
@@ -605,8 +605,10 @@ def mapIdx {α : Type u} {β : Type v} (f : Nat → α → β) (as : Array α) :
   Id.run <| as.mapIdxM f
 
 /-- Turns `#[a, b]` into `#[(a, 0), (b, 1)]`. -/
-def zipWithIndex (arr : Array α) : Array (α × Nat) :=
```

Comment:
Nitpick: As an alternative, have we considered keeping this as-is, and renaming the other function to `mapWithIndex`?

The long name feels more consistent with `zipWith` and `indexOf`.

Reply:
Though I see `eraseIdx` and `findIdx` so maybe that ship has sailed.

Reply:
In fact, it is inconsistent with `zipWith`, where I want the `with` to indicate that it is a higher order function, taking the combining function as an argument. This made us want to take the `with` *out*.

Separately, I'm intending to move more `index` to `idx` in future.